drivers/ipm/ipm_sedi.c

```diff
@@ -0,0 +1,310 @@
+/*
+ * Copyright (c) 2020-2023 Intel Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT intel_sedi_ipm
+
+#include <zephyr/kernel.h>
+#include <zephyr/init.h>
+#include <zephyr/device.h>
+#include <zephyr/pm/device.h>
+#include <zephyr/sys/sys_io.h>
+#include <errno.h>
+
+#include <zephyr/devicetree.h>
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(ipm_sedi, CONFIG_IPM_LOG_LEVEL);
+
+#ifdef CONFIG_SHARED_IRQ
+#include <zephyr/shared_irq.h>
+#endif
+
+#include <zephyr/drivers/ipm.h>
+#include "ipm_sedi.h"
+
+extern void sedi_ipc_isr(IN sedi_ipc_t ipc_device);
+
+static void set_ipm_dev_busy(const struct device *dev, bool is_write)
+{
+	struct ipm_sedi_context *ipm = dev->data;
+	unsigned int key = irq_lock();
```

Comment:
could irq_lock be removed?
irq_lock + atomic, seems weired.

Reply:
the var is defined as atomic, can only be set with this api